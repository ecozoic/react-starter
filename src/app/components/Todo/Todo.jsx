import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './Todo.scss';

const Todo = ({ onClick, completed, text }) => {
  const className = classNames(
    styles.todo,
    { [styles.completed]: completed },
  );

  return (
    <li
      className={className}
      onClick={onClick}
    >
      {text}!!
    </li>
  );
};

Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
};

export default Todo;
