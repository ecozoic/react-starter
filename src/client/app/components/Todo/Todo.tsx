import * as React from 'react';
import * as PropTypes from 'prop-types';
import * as classNames from 'classnames';

const styles = require('./Todo.scss');

export interface TodoProps {
  onClick: React.EventHandler<React.MouseEvent<HTMLLIElement>>;
  completed: boolean;
  text: string;
}

class Todo extends React.Component<TodoProps, undefined> {
  static propTypes = {
    onClick: PropTypes.func.isRequired,
    completed: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired,
  };

  render() {
    const { onClick, completed, text } = this.props;
    const className = classNames(
      styles.todo,
      { [styles.completed]: completed },
    );

    return (
      <li
        className={className}
        onClick={onClick}
      >
        {text}
      </li>
    );
  }
}

export default Todo;
