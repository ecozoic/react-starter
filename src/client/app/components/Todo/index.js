/* @flow */
export { default } from './Todo';
