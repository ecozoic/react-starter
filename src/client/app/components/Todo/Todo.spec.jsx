/* @flow */
import React from 'react';
import { mount, shallow } from 'enzyme';

import Todo from './Todo';

describe('Todo', () => {
  it('renders without crashing', () => {
    const todo = mount(
      <Todo
        onClick={function noop() { }}
        text={'Todo'}
        completed={false}
      />,
    );

    expect(todo).toMatchSnapshot();
  });

  describe('render', () => {
    it('displays todo text', () => {
      const todo = shallow(
        <Todo
          onClick={function noop() { }}
          text={'Todo'}
          completed={false}
        />,
      );

      const expected = 'Todo';
      const actual = todo.find('li').text();

      expect(actual).toBe(expected);
    });

    it('displays strikethrough if completed', () => {
      const todo = shallow(
        <Todo
          onClick={function noop() { }}
          text={'Todo'}
          completed
        />,
      );

      const expected = 'line-through';
      const actual = todo.props().style.textDecoration;

      expect(actual).toBe(expected);
    });

    it('displays no strikethrough if incomplete', () => {
      const todo = shallow(
        <Todo
          onClick={function noop() { }}
          text={'Todo'}
          completed={false}
        />,
      );

      const expected = 'none';
      const actual = todo.props().style.textDecoration;

      expect(actual).toBe(expected);
    });

    it('fires onclick when clicked', () => {
      const mockFn = jest.fn();

      const todo = shallow(
        <Todo
          onClick={mockFn}
          text={'Todo'}
          completed={false}
        />,
      );

      todo.find('li').simulate('click');

      expect(mockFn).toHaveBeenCalledTimes(1);
    });

    it('has todo class', () => {
      const todo = shallow(
        <Todo
          onClick={function noop() { }}
          text={'Todo'}
          completed={false}
        />,
      );

      expect(todo.find('li').hasClass('todo')).toBe(true);
    });
  });
});
