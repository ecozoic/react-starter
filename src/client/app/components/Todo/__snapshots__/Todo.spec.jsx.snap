// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders without crashing 1`] = `
<Todo
  completed={false}
  onClick={[Function]}
  text="Todo"
>
  <li
    data-radium={true}
    onClick={[Function]}
    style={
      Object {
        "fontSize": "18px",
        "userSelect": "none",
      }
    }
  >
    Todo
  </li>
</Todo>
`;
